crates/forge_app/src/tool_registry.rs

```diff
@@ -83,9 +91,15 @@ impl<S: Services> ToolRegistry<S> {
 
                 send_fs_search_context(self.services.as_ref(), context, &input, &output).await?;
 
-                Ok(crate::execution_result::ExecutionResult::from(output))
+                Ok(ExecutionResult::from(output))
             }
             Tools::ForgeToolFsRemove(input) => {
+                let _ = self
+                    .services
+                    .file_snapshot_service()
+                    .create_snapshot(Path::new(&input.path))
+                    .await?;
```

Comment:
Can create a a helper `self.snapshot(input.path)` instead of writing this boilerplate.